### PR TITLE
[WR186] Extract Request Body For middleware and replace

### PR DESCRIPTION
**Pull request by rosasck**
This extracts the body, places it in the context and then replaces the body for final use by handler 

**This was the nicest way to grab a body param in a middleware that i could find, all ideas to make this better welcomed and wanted**

### CODE REVIEW

File: server/middleware.go [hunk 1]
```diff
@@ -206,3 +209,27 @@ func TrimSlash(h http.Handler) http.Handler {
...
+		var err error
+		rawBody, err = ioutil.ReadAll(body)
+		if err != nil {
+			w.WriteHeader(http.StatusRequestEntityTooLarge)
+			return
+		}
+		// Repopulate the request body for the ultimate consumer of this request
+		r.Body = ioutil.NopCloser(&rawBodyBuffer)
+	}
+	rawBodyContext := context.WithValue(r.Context(), KeyRawBody, rawBody)
```
Comment (efabens) [inline server/middleware.go]: I don't think that the `rawBodyContext` can be consumed by downstream. It don't believe it is added to the request

**Comment by rosasck** [comment]
After a quick chat with Eli, he was able to explain chaining Middlewares with me

**Review by efabens — APPROVED** [review]
Next time we are in cyclops we should switch to using this implementation

File: server/middleware.go [hunk 2]
```diff
@@ -23,6 +24,8 @@ const (
 	HealthCheckPathSuffix = "/healthcheck"
 	// ServiceCheckPathSuffix is a centrally defined service check path.
 	ServiceCheckPathSuffix = "/servicecheck"
+	// KeyRawBody is the context key for the raw body value in a request context
+	KeyRawBody = "rawBody"
```
Comment (galxy25) [inline server/middleware.go]: `RawBodyContextKey`

**Review by galxy25 — APPROVED** [review]
🎉

File: go.mod [hunk 3]
```diff
@@ -19,4 +19,4 @@ require (
 	mellium.im/sasl v0.2.1 // indirect
```
Comment (rosasck) [inline go.mod:19]: @efabens i updated from 1.13 is there anything else i need to do

**Pull request merged**